terminology/cbm.md

```diff
@@ -1,46 +1,71 @@
+-   detects incompatible data types passed to it and _throws_ useful error messages  (that,  then,  get handled outside the CBM)
+-   renders performantly  (as a lot of CBMs might get rendered simultaneously)
+-   is free of any external resource dependencies,  i.e.,  all a CBM needs to render is the data provided to it in a suitable JSONLike format.
+- has as much localization as possible, but at least provides English information
 
-The component is the CBM shown to the user and must work for widths from 320px to 1080px. It gets passed the following `props`:
+## Technical requirements
+### Distribution and the `.cbm` file format
+CBMs get distributed via a CBM registry. A central, most-used official registry, provided by the SerSta team, exists, but it's also possible to host one's own registry, e.g., for internal company purposes.
 
-*   `data: JSONSerializable`: The data returned by the API endpoint
-*   `onRegisterAction: (name, action) => void` a function that lets the CBM register an action for the "Three-Dots/More Actions" menu
-*   `onRefresh: () => Promise<void>` a function to reload the API response for this CBM from inside the CBM (e.g., for modules requiring frequent updates etc.)
+To upload a CBM into the registry, a `.cbm` file has to get built using the `@server-state/cli`, available in npm.
 
-Components get registered in `/src/config/component-registry.js` as the object importable from `my-cbm`, i.e., the object containing `component` and `info`. Because of this, it's possible to use something like:
+This file format is, at a basic level, a JSON file containing an `object` with the following properties:
+- `id: string` A unique plugin id, provided by the registry.
+- `name: string` The CBM's name
+- `version: string` An npm-compatible version number
+- `description: string` A description of the CBM
+- `support_url: string` A url where users can get support, including the capability of contacting the CBM's author
+- `website: string | undefined` The CBM's website, if applicable
+- `repo_url: string | undefined` The CBM's repository, if applicable
+- `code: string` The CBM's code in JS, see the specification below
 
-```
-import MyCbm from 'my-cbm';
+### Specification of the `.cbm`'s `code: string` property
+The `code` property is a string containing valid JavaScript code that exports (in CommonJS2, i.e., using `module.exports = {}`) an object containing the following properties:
 
-const components = {
-  MyCbm.info.name: MyCbm,
-  [...]
-};
-```
+- `component: React.Component` A react component that actually defines the CBM, as defined below in *The React Component*
+- `isCompatible: (data: JSONSerializable) => boolean | undefined` A function that checks whether the `data` is in a format that can get visualized by this CBM. For compatibility reasons, providing this function is optional, but not providing it leads to a flag stating *Compatibility unknown* in the CBM selection screen in the *client-base*.
+
```

Comment:
Is the logoURL missing in the object? (see: https://github.com/server-state/template-cbm/blob/master/src/index.js)

Reply:
Logos will get managed by the CBM registry in the future. Therefore, this is no longer required here and also not practical since at the time of browsing CBMs in the "CBM Manager", we wouldn't have access to this, if we did it here 🙂